tests/mock.fish

```diff
@@ -18,7 +18,7 @@
 #
 
 function mock -a cmd -a argument -a exit_code -a executed_code -d "Mock library for fish shell testing"
-	set -l cmd_blacklist "builtin" "functions" "eval" "command"
+	set -l cmd_blacklist "builtin" "functions" "eval" "command" "argparse" "read" "set" "status" "test"
```

Comment:
This is missing `[`, the alternate name for `test`. Though, technically, there's a few more. See https://github.com/fish-shell/fish-shell/blob/master/src/parser_keywords.cpp. So the list is

```fish
"[" and argparse begin break builtin case command continue else end exec for function if not or read return set status switch test while
```

(yes, `[` is the only one that needs to be quoted)

Reply:
This seems like a pain in the butt to maintain as things are added over time. Can fish do this for us? Is there a command to list all builtins automatically?